### PR TITLE
Fix regular expressions and implement delete

### PR DESCRIPTION
**Pull request by e-kwsm**

### CODE REVIEW

**Comment by Shougo** [comment]
Please describe why `\S` is needed?

File: rplugin/python3/deoppet/parser.py [hunk 1]
```diff
@@ -43,25 +43,29 @@ def parse(self, text: str) -> Snippet:
...
-            m = re.search(r'^\s*delete\s+(.*)$', line)
+            m = re.search(r'^\s*delete\s+(\S.*)$', line)
             if m:
                 # Delete snippet trigger
-                # Todo
+                s = m.group(1)
+                try:
+                    del snippets[s]
+                except KeyError:
+                    self.error(f'try to delete a non-existing snippet {s}')
```
Comment (Shougo) [inline rplugin/python3/deoppet/parser.py:53]: Please check it by `if s in snippets`.

**Comment by e-kwsm** [comment]
> Please describe why `\S` is needed?

For example, `r'^\s*snippet\s+(.*)$'` matches `snippet `, while `r'^\s*snippet\s+(\S.*)$'` doesn't. If a snippet is not allowed to have spaces, the regex can be `r'^\s*snippet\s+(\S+)`.

When the snippet exists, 
```
Error detected while processing command line:
error caught in async handler '~/.local/share/nvim/site/bundle/.cache/init.vim/.dein/rplugin/python3/deoppet:function:_deoppet_init [[]]'
Traceback (most recent call last):
  File "~/.local/share/nvim/site/bundle/.cache/init.vim/.dein/rplugin/python3/deoppet/__init__.py", line 31, in init_channel
    self._deoppet = Deoppet(self._vim)
  File "~/.local/share/nvim/site/bundle/.cache/init.vim/.dein/rplugin/python3/deoppet/deoppet.py", line 28, in __init__
    self._load_snippets()
  File "~/.local/share/nvim/site/bundle/.cache/init.vim/.dein/rplugin/python3/deoppet/deoppet.py", line 76, in _load_snippets
    buf.vars['deoppet_snippets'] = snippets
  File "~/.local/lib/python3.7/site-packages/pynvim/api/common.py", line 116, in __setitem__
    self._set(key, value)
  File "~/.local/lib/python3.7/site-packages/pynvim/api/common.py", line 58, in request
    return self._session.request(name, self, *args, **kwargs)
  File "~/.local/lib/python3.7/site-packages/pynvim/api/nvim.py", line 182, in request
    res = self._session.request(name, *args, **kwargs)
  File "~/.local/lib/python3.7/site-packages/pynvim/msgpack_rpc/session.py", line 104, in request
    raise self.error_wrapper(err)
pynvim.api.common.NvimError: Empty dictionary keys aren't allowed
```

**Comment by Shougo** [comment]
OK.

**Pull request merged**